driver/src/main/java/org/neo4j/driver/internal/net/pooling/SocketConnectionPool.java

```diff
@@ -68,26 +71,41 @@ public SocketConnectionPool( PoolSettings poolSettings, Connector connector, Clo
     public Connection acquire( final BoltServerAddress address )
     {
         assertNotClosed();
+        BlockingPooledConnectionQueue connectionQueue = pool( address );
+        PooledConnection connection = acquireConnection( address, connectionQueue );
```

Comment:
It could also simplify things around these suppliers and consumers. Let's think about it.